### ISSUE TITLE
E3002 rule triggered when using Macros

### ISSUE THREAD

**Issue by khan-a1** [aws-cloudformation/cfn-lint#2573]
### CloudFormation Lint Version

cfn-lint 0.72.10

### What operating system are you using?

Mac

### Describe the bug

I have a Macro that I use to generate dynamic resource Tags. It is a valid template and deploys fine to my AWS account using AWS CLI. But when I run cfn-lint, it triggers E3002 rule. I can suppress it using the metadata but it should not trigger this rule in the first place

### Expected behavior

It should pass as valid template and not trigger E3002

### Reproduction template

```yaml

AWSTemplateFormatVersion: 2010-09-09
Resources:
  S3Bucket:
    Type: "AWS::S3::Bucket"
    Properties:
      Tags:
        "Fn::Transform":
          - Name: "TaggingMacro"
            Parameters:
              SSMParameterName: "/apps/dcs/crm"
              TagOne: Example
              TagTwo: World
```

**Comment by PatMyron** [comment]
tracking in https://github.com/aws-cloudformation/cfn-lint/issues/476
